legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java

```diff
@@ -124,6 +124,107 @@ private static DataSpaceExecutableResult buildExecutableResult(ResultType result
+                                            info.title = executable.title;
+                                            info.description = executable.description;
+                                            info.type = PackageableElementType.SERVICE.name();
+                                            info.query = ((PureSingleExecution) ((Service) v1Service.get()).execution).func;
+                                            return info;
+                                        }
+                                    }
+                                    else if (((Root_meta_legend_service_metamodel_Service) service)._execution() instanceof Root_meta_legend_service_metamodel_PureMultiExecution_Impl)
+                                    {
+                                        Root_meta_legend_service_metamodel_PureMultiExecution multiExecution = (Root_meta_legend_service_metamodel_PureMultiExecution) ((Root_meta_legend_service_metamodel_Service) service)._execution();
+                                        boolean qualifiedService = true;
+                                        for (Root_meta_legend_service_metamodel_KeyedExecutionParameter keyedParam : multiExecution._executionParameters())
+                                        {
+                                            String mappingPath = platform_pure_essential_meta_graph_elementToPath.Root_meta_pure_functions_meta_elementToPath_PackageableElement_1__String_1__String_1_(keyedParam._mapping(), "::", pureModel.getExecutionSupport());
+                                            String runtimePath = pureModel.getRuntimePath(keyedParam._runtime());
+                                            if (!(qualifiedMappingPath.contains(mappingPath) && qualifiedRuntimePath.contains(runtimePath)))
+                                            {
+                                                qualifiedService = false;
+                                            }
+                                        }
+                                        if (qualifiedService)
+                                        {
+                                            DataSpaceExecutableAnalysisInfo info = new DataSpaceExecutableAnalysisInfo();
+                                            info.executionContextKey = executable.executionContextKey;
+                                            info.title = executable.title;
+                                            info.description = executable.description;
+                                            info.type = PackageableElementType.SERVICE.name();
+                                            info.query = ((PureMultiExecution) ((Service) v1Service.get()).execution).func;
+                                            return info;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        catch (Exception ignored)
```

Comment:
will we even ever reach here if the above does not fail i.e not a service ?

Reply:
the path used in the function pointer is not the actual function path, you will get an except if you try to get the packageelement using this path 
instead, you should generate the correct function path first, then get the v1-function